src/core/infer_request.cc

```diff
@@ -181,49 +275,50 @@ InferenceRequest::Normalize(const InferenceBackend& backend)
   for (auto& pr : inputs_) {
     const ModelInput* input_config;
     RETURN_IF_ERROR(backend.GetInput(pr.first, &input_config));
+    auto& input = pr.second;
 
-    auto& shape = *pr.second.MutableShape();
+    auto new_shape = input.MutableShape();
+    *new_shape = input.OriginalShape();
```

Comment:
This should be moved to the if clause in line 332? Otherwise that if predicate will always be false.

Reply:
OriginalShape will be empty if the request didn't specify a shape.